### PR TITLE
Bump `flow-go` version

### PR DESCRIPTION
**Pull request by jordanschalm**
## Description

Updates the pinned `flow-go` version to latest `master` (`v0.17.6`) in order to address recent breaking changes in `fvm`. These breakages are currently a blocker for the epochs feature branch on `flow-go`.

⚠️ I need some help with the configuration change required here. I'm not sure how to convert between the context config and the bootstrap config for restricted account deployment.

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: blockchain.go [hunk 1]
```diff
@@ -441,6 +440,7 @@ func configureBootstrapProcedure(conf config, flowAccountKey flowgo.AccountPubli
 	options := make([]fvm.BootstrapProcedureOption, 0)
 	options = append(options,
 		fvm.WithInitialTokenSupply(supply),
+		fvm.WithRestrictedAccountCreationEnabled(false),
```
Comment (jordanschalm) [inline blockchain.go:443]: Tests fail with `missing account created event` with only this change.

**Comment by turbolent** [comment]
> I need some help with the configuration change required here. I'm not sure how to convert between the context config and the bootstrap config for restricted account deployment.

@jordanschalm Can you elaborate on the problem? As in, what is the issue when using the Emulator without the configuration changes?

File: blockchain.go (hunk 1 above)
Comment (janezpodhostnik) [reply blockchain.go:443]: I can take a look. I know what the problem is!

**Comment by janezpodhostnik** [comment]
The problem is that account creation emits more than one event now. The emulator tests assumed that there is only one event. This same assumption might be problematic in the go-sdk examples (and maybe elsewhere) this should be communicated as a breaking change. I will pass this on.

**Comment by turbolent** [comment]
Oh, I think I understand now, thanks for the explanation 👍

**Review by turbolent — APPROVED** [review]
Can we maybe even bump to today's Canary spork version which has Cadence v0.17.0?

**Comment by jordanschalm** [comment]
Hey @turbolent I updated again to the latest version. There was one small change from the Cadence version bump - could you take another look before I merge?

**Review by turbolent — APPROVED** [review]
Looks good, thank you for updating this! 👍

**Pull request merged**